src/charm.py

```diff
@@ -149,25 +159,22 @@ def _get_interfaces(self):
             raise ErrorWithStatus(err, BlockedStatus)
         return interfaces
 
-    def _get_relational_db_data(
-        self,
-    ):
-        relational_db = self.model.relations["relational-db"]
-        if len(relational_db) > 1:
-            raise ErrorWithStatus(
-                f"Too many mysql relations. Found {len(relational_db)}, expected 1", BlockedStatus
-            )
-
-        try:
-            relational_db = relational_db[0]
-            db_unit = list(relational_db.units)[0]
-            relational_db = relational_db.data[db_unit]
-            relational_db["database"]
-            return relational_db
-        except IndexError:
-            raise ErrorWithStatus("Waiting for relational-db relation data", WaitingStatus)
-        except KeyError:
-            raise ErrorWithStatus("Missing data in relational-db relation", WaitingStatus)
+    def _get_relational_db_data(self) -> dict:
+        data = self.database.fetch_relation_data()
+        self.logger.debug("Got following database data: %s", data)
+        for _, val in data.items():
+            if not val:
+                continue
+            self.logger.info("New mysql database endpoint is %s", val["endpoints"])
+            host, port = val["endpoints"].split(":")
+            db_data = {
+                "host": host,
+                "port": port,
+                "username": val["username"],
+                "password": val["password"],
+            }
+            return db_data
+        raise ErrorWithStatus("Waiting for relational-db relation data", WaitingStatus)
```

Comment:
what will happen if relation does not exist ?

can we identify that there is no relation and ask to add it?

Reply:
well it will set status to waiting with `Waiting for relational-db relation data`. Do you want to change the text to sth else ?

Reply:
Should we have Blocked status in this case with message `Please add required relational-db relation`?

Reply:
I agree, we should add more logging/status setting:
- Relation does not exist: Blocked/Please add . . .
- Data is empty: Waiting/Found empty relation data  . . .
- Invalid data: Waiting/Found incorrect data  . . .

Reply:
Added